### DISCUSSION TITLE
mintDigitalAssetTransaction behaviour

### DISCUSSION THREAD

**Discussion by bjmvercelli** [aptos-labs/aptos-developer-discussions#200] [Questions]
### Discord user ID

361196456922841091

### Describe your question in detail.

Hey, how you guys doing?

I was trying to automate a mint proccess and came up with `mintDigitalAssetTransaction` function, but couldn't figure it out

You can check the correct transaction [here](https://explorer.aptoslabs.com/txn/601839451?network=mainnet).

I was trying to generate my transaction like this:
```ts
 const transaction = await aptos.digitalAsset.mintDigitalAssetTransaction({
    collection: "0x39673a89d85549ad0d7bef3f53510fe70be2d5abaac0d079330ade5548319b62",
    description: "The first community initiative to mint 1 Million NFT's on Aptos",
    name: "It's an Aptos thing",
    uri: "https://arweave.net/SjHGrv4DZQtAJdToDsn7N3yJSiH1WrxM3tYECkspiXk",
    creator: account,
  })
```

The result was this failed [transaction](https://explorer.aptoslabs.com/txn/601843783?network=mainnet)

Where was my mistake?

### What have you tried or looked at?  Or how can we reproduce the error?

I've tried to look at docs

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by 0xmaayan** [comment]
As stated in the function doc, the `collection` should be `the name of the collection the digital asset belongs to`
https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/api/digitalAsset.ts#L288
